build-utils/src/index.ts

```diff
@@ -0,0 +1,149 @@
+import path, { resolve as pathResolve } from 'path';
+import { createReadStream, stat, mkdtemp, exists } from 'mz/fs';
+import { mkdirp, remove, copy } from 'fs-extra';
+import { tmpdir } from 'os';
+import tar from 'tar';
+import fetch from 'node-fetch';
+import shellEcape from 'any-shell-escape';
+import { spawn } from '@reshuffle/utils-subprocess';
+import { getDependencies, MismatchedPackageAndPackageLockError } from './getdeps';
+
+export { MismatchedPackageAndPackageLockError };
+
+export interface Logger {
+  log: (str: string) => void;
+  error: (str: string) => void;
+}
+
+export interface BuildOptions {
+  skipNpmInstall: boolean;
+  logger: Logger;
+}
+
+const DEFAULT_OPTIONS: BuildOptions = {
+  skipNpmInstall: false,
+  logger: console,
+};
+
+function escapeWin32(filePath: string) {
+  return process.platform === 'win32' ? shellEcape(filePath) : filePath;
+}
+
+export async function build(projectDir: string, options?: Partial<BuildOptions>): Promise<string> {
+  const {
+    skipNpmInstall,
```

Comment:
Nit: this is already new code and not moved old code

Reply:
correct
is this a problem?